core/validator.go

```diff
@@ -43,74 +108,132 @@ func (v *chunkValidator) ValidateBlob(blob *BlobMessage, operatorState *Operator
+}
 
-		chunkLength, err := v.assignment.GetChunkLengthFromHeader(operatorState, quorumHeader)
-		if err != nil {
-			return err
-		}
+func (v *chunkValidator) UpdateOperatorID(operatorID OperatorID) {
+	v.operatorID = operatorID
+}
 
-		// Validate the chunkLength against the quorum and adversary threshold parameters
-		numOperators := uint(len(operatorState.Operators[quorumHeader.QuorumID]))
-		minChunkLength, err := v.assignment.GetMinimumChunkLength(numOperators, blob.BlobHeader.BlobCommitments.Length, quorumHeader.QuantizationFactor, quorumHeader.QuorumThreshold, quorumHeader.AdversaryThreshold)
-		if err != nil {
-			return err
-		}
-		params, err := GetEncodingParams(minChunkLength, info.TotalChunks)
-		if err != nil {
-			return err
-		}
+func (v *chunkValidator) ValidateBatch(blobs []*BlobMessage, operatorState *OperatorState, pool common.WorkerPool) error {
+	subBatchMap := make(map[EncodingParams]*SubBatch)
+	blobCommitmentList := make([]BlobCommitments, len(blobs))
 
-		if params.ChunkLength != chunkLength {
-			return errors.New("number of chunks does not match assignment")
+	for k, blob := range blobs {
+		if len(blob.Bundles) != len(blob.BlobHeader.QuorumInfos) {
+			return errors.New("number of bundles does not match number of quorums")
 		}
 
-		// Get the chunk length
-		chunks := blob.Bundles[quorumHeader.QuorumID]
-		for _, chunk := range chunks {
-			if uint(chunk.Length()) != chunkLength {
-				return ErrChunkLengthMismatch
+		// Validate the blob length
+
+		blobCommitmentList[k] = blob.BlobHeader.BlobCommitments
+
+		// for each quorum
+		for _, quorumHeader := range blob.BlobHeader.QuorumInfos {
+			// Check if the operator is a member of the quorum
+			chunks, assignment, params, err := v.preprocessBlob(quorumHeader, blob, operatorState)
+			if err == ErrBlobQuorumSkip {
+				continue
+			} else if err != nil {
+				return err
+			} else {
+				// Check the received chunks against the commitment
+				blobIndex := 0
+				subBatch, ok := subBatchMap[*params]
+				if ok {
+					blobIndex = subBatch.NumBlobs
+				}
+
+				indices := assignment.GetIndices()
+				samples := make([]Sample, len(chunks))
+				for ind := range chunks {
+					samples[ind] = Sample{
+						Commitment:      blob.BlobHeader.BlobCommitments.Commitment,
+						Chunk:           chunks[ind],
+						AssignmentIndex: uint(indices[ind]),
+						BlobIndex:       blobIndex,
+					}
+				}
+
+				// update subBatch
+				if !ok {
+					subBatchMap[*params] = &SubBatch{
+						Samples:  samples,
+						NumBlobs: 1,
+					}
+				} else {
+					subBatch.Samples = append(subBatch.Samples, samples...)
+					subBatch.NumBlobs += 1
+				}
 			}
 		}
+	}
 
-		// Validate the chunk length
-		if chunkLength*quorumHeader.QuantizationFactor*numOperators != quorumHeader.EncodedBlobLength {
-			return ErrInvalidHeader
-		}
+	// Parallelize the universal verification for each subBatch
+	numSubBatch := len(subBatchMap)
+	numResult := numSubBatch + len(blobCommitmentList)
```

Comment:
nit: numSubBatch is just used here, so inline it (`numResult := len(numSubBatch) + len(blobCommitmentList)`)